pkg/vendir/fetch/git/git.go

```diff
@@ -93,7 +93,9 @@ func (t *Git) fetch(dstPath string, tempArea ctlfetch.TempArea) error {
 		if authOpts.PrivateKey != nil {
 			path := filepath.Join(authDir, "private-key")
 
-			err = os.WriteFile(path, []byte(*authOpts.PrivateKey), 0600)
+			// Adding newlinne as git does not support private-key without newline as last character.
+			// Now vendir will work with both with/without newline in private-key.
+			err = os.WriteFile(path, []byte(*authOpts.PrivateKey+"\n"), 0600)
```

Comment:
At first, I noticed a typo (`newlinne`), but decided to expand the suggestion. How about this?

```suggestion
			// Ensure the private key ends with a newline character, as git requires it to work.
			err = os.WriteFile(path, []byte(*authOpts.PrivateKey+"\n"), 0600)
```